src/coreclr/nativeaot/Common/src/System/Collections/Generic/LowLevelDictionary.cs

```diff
@@ -1,304 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System;
-using System.Collections;
-using System.Collections.Generic;
-
-namespace System.Collections.Generic
-{
-    /*============================================================
-    **
-    ** Class:  LowLevelDictionary<TKey, TValue>
-    **
-    ** Private version of Dictionary<> for internal System.Private.CoreLib use. This
-    ** permits sharing more source between BCL and System.Private.CoreLib (as well as the
-    ** fact that Dictionary<> is just a useful class in general.)
-    **
-    ** This does not strive to implement the full api surface area
-    ** (but any portion it does implement should match the real Dictionary<>'s
-    ** behavior.)
-    **
-    ===========================================================*/
-    internal class LowLevelDictionary<TKey, TValue> where TKey : IEquatable<TKey>
```

Comment:
`Dictionary<TKey, TValue>` has been optimized for speed and have about twice of size impact than LowLevelDictionary. Although it can reduced by sharing generic instantiations, the rare instantiations (other than `Dictionary<__Canon, __Canon>`) will be pure overhead. Using `Dictionary` more can also amplify any size impact in the shared implementation, which we are aggressively optimizing for speed.

Are we interested to keep a "size-optimized Dictionary" implementation?